### PR TITLE
Fix metric setup for finetuning without a test set

### PR DESCRIPTION
**Pull request by MaximumEntropy**
Signed-off-by: MaximumEntropy <sandeep.subramanian.1@umontreal.ca>

# What does this PR do ?

Fixes a bug in setup_metric() when fine-tuning without a specified test set.

**Collection**: NLP

# Changelog 
- Use the provided data_cfg to instantiate a metric rather than test_ds.

# Usage
* N/A

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by okuchaiev — DISMISSED** [review]
lgtm

File: examples/nlp/language_modeling/megatron_t5_seq2seq_eval.py [hunk 1]
```diff
@@ -65,9 +65,14 @@ def main(cfg) -> None:
             trainer.callbacks[idx] = StatelessTimer(cfg.trainer.max_time,)
 
     # Get the T5 Base configuration.
-    t5_cfg = MegatronT5GLUEModel.restore_from(
-        restore_path=cfg.model.restore_from_path, trainer=trainer, return_config=True
-    )
+    if hasattr(t5_cfg.data.validation_ds, 'task_name'):
+        t5_cfg = MegatronT5GLUEModel.restore_from(
```
Comment (michalivne) [inline examples/nlp/language_modeling/megatron_t5_seq2seq_eval.py:69]: Do we need the correct model for restore_from when getting config? Can't we use `ModelPT` for both?

**Review by michalivne — COMMENTED** [review]
LGTM. See minor comment

File: examples/nlp/language_modeling/megatron_t5_seq2seq_eval.py (hunk 1 above)
Comment (MaximumEntropy) [reply examples/nlp/language_modeling/megatron_t5_seq2seq_eval.py:69]: I'd rather keep it this way because we don't have a CI/unit test yet for this eval script.

**Review by okuchaiev — APPROVED** [review]
lgtm
